cmake/CompilerChecks.cmake

```diff
@@ -23,11 +23,67 @@ if(NOT DEFINED CMAKE_INTERPROCEDURAL_OPTIMIZATION)
     endif()
 endif()
 
-# Check for CPUID headers
-try_compile(HAVE_CPUID ${PROJECT_BINARY_DIR}
-            ${PROJECT_SOURCE_DIR}/cmake/check_cpuid.cpp)
```

Comment:
The big advantage of the check file is that it checked in exactly the same way as the use in hsCpuID.h...  By pulling this into CMake, the check is no longer the same, which may result in odd errors when someone uses different compilers or tools :/

Reply:
I could pull the cpuid test back out into its own file, but I think the various simd header checks are good like this.

Reply:
Yeah, the SIMD tests should definitely be kept in CMake; I was just looking specifically at the cpuid check, which is more convoluted.

Reply:
Note for posterity, the SIMD stuff was split off to #862.